design/design-revised.md

```diff
@@ -0,0 +1,63 @@
+Teleport coding challenge design document by Jin Huang
+
+I) Architecture:
+
+The library will consist of the worker and worker service. At a high level view, a worker will be able
+to perform start, stop, and query status for a command, and the worker service provide a map 
+for the client to have control over multiple workers. The worker service also provides the workers a
+custom logger, which will be used to log output of a process (sdtout and stderr) during execution of a 
+running job to a file or a io.Writer.
+
+The structure will be as follows:
+
+Worker:
+- struct: UID, command, status, error, *logger, mutex
```

Comment:
How/where will the job's logs be stored?

Reply:
The job's output will be logged to a file using the passed in logger and the normal error and messages will be logged to the console with the standard logger.